src/components/svg-canvas-graph.tsx

```diff
@@ -105,6 +106,9 @@ const SvgCanvas = () => {
         [svgViewBoxMin, svgViewBoxZoom, width, height, pointerPosition]
     );
 
+    // the active snap points (only a pair of them and length == 2)
+    const [activeSnapPoints, setActiveSnapPoints] = React.useState<SnapPoint | undefined>(undefined);
```

Comment:
Could it be rewritten to `[SnapPoint, SnapPoint] | undefined`?

Reply:
Maybe the comment should be rewrote 🤔

Reply:
It's okay if 2 means the originalNodesPos in SnapPoint. Just explain the comment here better than `only a pair of them and length == 2` should suffice.